test/components/connect.spec.js

```diff
@@ -949,6 +949,38 @@ describe('React', () => {
       expect(decorated.refs.wrappedInstance.someInstanceMethod()).toBe(someData)
     })
 
+    it('should do no parsing when the response body is empty', (done) => {
+      window.fetch = (request) => {
+        return new Promise((resolve) => {
+          resolve(new window.Response('', { status: 204 }))
+        })
+      }
+
+      @connect(() => ({ testFetch: `/empty` }))
+      class Container extends Component {
+        render() {
+          return <Passthrough {...this.props} />
+        }
+      }
+
+      const container = TestUtils.renderIntoDocument(
+        <Container />
+      )
+
+      const init = TestUtils.findRenderedComponentWithType(container, Container)
+      expect(init.state.data.testFetch).toIncludeKeyValues(
+        { fulfilled: false, pending: true, reason: null, refreshing: false, rejected: false, settled: false, value: null }
+      )
+
+      setImmediate(() => {
+        const fulfilled = TestUtils.findRenderedComponentWithType(container, Container)
+        expect(fulfilled.state.data.testFetch).toIncludeKeyValues(
+          { fulfilled: true, pending: false, reason: null, refreshing: false, rejected: false, settled: true, value: null }
+        )
+        done()
+      })
+    })
+
```

Comment:
I did my best to strip this down to the essential test, but I feel like it could be stripped even more. Also the error message is not fantastic (buried in the diff provided by `toIncludeKeyValues`), so that could probably be improved.